pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,282 @@
+
+        # Base directory for output and counter reset
+        self.basedir = self.cfg.getpath(cfgsect, 'basedir', '.', abs=True)
+        self.nout = {}
+        for sname, scfg in self._scene_cfg.items():
+            name = scfg['image_name']
+            self.nout[name] = file_index(self.basedir, name) if intg.isrestart else 0
+
+        # Set subdivision
+        self.divisor = self.cfg.getint(cfgsect, 'division', 
+                                       self.cfg.getint('solver', 'order'))
+        self.etypes_div = defaultdict(lambda: self.divisor)
+
+        # Share number of ele types per part
+        ndoms = len(system.ele_map)
+        ndoms = comm.gather(ndoms, root=root)
+        if rank == root:
+            ndoms = np.asarray(ndoms)
+        ndoms = comm.bcast(ndoms, root=root)
+        self.dom_0 = dom_0 = np.sum(ndoms[:rank])
+
+        # Generate conduit blueprint of mesh
+        mesh_n = ConduitNode(self.conduit)
+        self.ops = {}
+        for i, (ename, eles) in enumerate(system.ele_map.items(), start=dom_0):
+            if ename in ['pri', 'pyr']:
+                raise ValueError(f'Ascent: {ename} elements not supported')
+
+            dpf = f'domain_{i}/'
+
+            shape = subclass_where(BaseShape, name=ename)(eles.nspts, self.cfg)
+
+            svpts = shape.std_ele(self.etypes_div[ename])
+            nsvpts = len(svpts)
+
+            mesh_op = shape.sbasis.nodal_basis_at(svpts).astype(self.dtype)
+            soln_op = shape.ubasis.nodal_basis_at(svpts).astype(self.dtype)
+            self.ops[ename] = (mesh_op, soln_op)
+
+            mesh_n[f'{dpf}state/domain_id'] = rank
+            mesh_n[f'{dpf}coordsets/coords/type'] = 'explicit'
+            mesh_n[f'{dpf}topologies/mesh/coordset'] = 'coords'
+            mesh_n[f'{dpf}topologies/mesh/type'] = 'unstructured'
+            mesh_n[f'{dpf}topologies/mesh/elements/shape'] = ename
+
+            subdvcls = subclass_where(BaseShapeSubDiv, name=ename)
+            sconn = subdvcls.subnodes(self.divisor)
+            subdvcon = np.hstack([sconn + j*nsvpts for j in range(eles.neles)])
+            mesh_n[f'{dpf}topologies/mesh/elements/connectivity'] = subdvcon
+
+            xd = np.einsum('ik,k...', mesh_op, eles.ploc_at_np('linspts'))
+            for l , x in zip('xyz', xd.reshape(eles.ndims, -1)):
+                mesh_n[f'{dpf}coordsets/coords/values/{l}'] = x
+
+            for field in self._expr:
+                mesh_n[f'{dpf}fields/{field}/association'] = 'vertex'
+                mesh_n[f'{dpf}fields/{field}/volume_dependent'] = 'false'
+                mesh_n[f'{dpf}fields/{field}/topology'] = 'mesh'
+
+        self.mesh_n = mesh_n
+
+        # Inital Ascent setup
+        self.a = Ascent()
+        self.ascent_config = ConduitNode(self.conduit)
+        self.ascent_config['mpi_comm'] = comm.py2f()
+        self.ascent_config['runtime/type'] = 'ascent'
+        self.ascent_config['runtine/vtkm/backend'] = 'serial'
```

Comment:
Is this something we want to make an option?

Reply:
I'm not sure, I can expose it, but default to serial